src/components/Modal/Modal.tsx

```diff
@@ -0,0 +1,51 @@
+import styles from './Modal.module.scss';
+import { PropsWithChildren } from 'react';
+
+interface ModalProps {
+  h3: string;
+  onClose: () => void;
+  onCanсel: () => void;
+  onSave: () => void;
+}
+
+export default function Modal(props: PropsWithChildren<ModalProps>) {
+  return (
+    <>
+      <div
+        className={styles.modalcontainer}
+        onClick={(e) => {
+          if (e.currentTarget.className === styles.modalcontainer) {
+            e.stopPropagation();
+            props.onClose();
+          }
```

Comment:
```suggestion
          props.onClose();
```

you probably don't need the condition. should be just props.onClose. the event bubbling is being stopped in the div below, so _this_ handler won't be called anyway (if click was on that div)

Reply:
understood sir